pkg/service/storage/ucan.go

```diff
@@ -172,6 +179,100 @@ func NewUCANServer(storageService Service, options ...server.Option) (server.Ser
 				},
 			),
 		),
+		server.WithServiceMethod(
+			replica.AllocateAbility,
+			server.Provide(
+				replica.Allocate,
+				func(cap ucan.Capability[replica.AllocateCaveats], inv invocation.Invocation, iCtx server.InvocationContext) (replica.AllocateOk, fx.Effects, error) {
+					//
+					// UCAN Validation
+					//
+
+					// only service principal can perform an allocation
+					if cap.With() != iCtx.ID().DID().String() {
+						return replica.AllocateOk{}, nil, NewUnsupportedCapabilityError(cap)
+					}
+
+					//
+					// end UCAN Validation
+					//
+
+					// create the transfer invocation: an fx of the allocate invocation receipt.
+					trnsfInv, err := replica.Transfer.Invoke(
+						storageService.ID(),
+						storageService.ID(),
+						storageService.ID().DID().GoString(),
+						replica.TransferCaveats{
+							Space:    cap.Nb().Space,
+							Blob:     cap.Nb().Blob,
+							Location: cap.Nb().Location,
+							Cause:    inv.Link(),
+						},
+					)
+					if err != nil {
+						return replica.AllocateOk{}, nil, failure.FromError(err)
+					}
+					for block, err := range inv.Blocks() {
+						if err != nil {
+							return replica.AllocateOk{}, nil, fmt.Errorf("iterating replica allocate invocation blocks: %w", err)
+						}
+						if err := trnsfInv.Attach(block); err != nil {
+							return replica.AllocateOk{}, nil, fmt.Errorf("failed to replica allocate invocation block (%s) to transfer invocation: %w", block.Link().String(), err)
+						}
+					}
```

Comment:
to reviewer: is this the correct way to chain the replica/allocate invocation to the replica/transfer invocation?